fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedCompoundReader.java

```diff
@@ -0,0 +1,196 @@
+            Throwable priorE = null;
+            try {
+                CodecUtil.checkIndexHeader(entriesStream, LuceneOptimizedCompoundFormat.ENTRY_CODEC,
+                         LuceneOptimizedCompoundFormat.VERSION_START,
+                         LuceneOptimizedCompoundFormat.VERSION_CURRENT, segmentID, "");
+                final int numEntries = entriesStream.readVInt();
+                mapping = new HashMap<>(numEntries);
+                for (int i = 0; i < numEntries; i++) {
+                    final FileEntry fileEntry = new FileEntry();
+                    final String id = entriesStream.readString();
+                    FileEntry previous = mapping.put(id, fileEntry);
+                    if (previous != null) {
+                        throw new CorruptIndexException("Duplicate cfs entry id=" + id + " in CFS ", entriesStream);
+                    }
+                    fileEntry.offset = entriesStream.readLong();
+                    fileEntry.length = entriesStream.readLong();
+                }
+            } catch (Throwable exception) {
+                priorE = exception;
+            } finally {
+                CodecUtil.checkFooter(entriesStream, priorE);
+            }
+        }
+        return Collections.unmodifiableMap(mapping);
+    }
+
+    @Override
+    public void close() throws IOException {
+        IOUtils.close(handle);
+    }
+
+    @Override
+    public IndexInput openInput(String name, IOContext context) throws IOException {
+        ensureOpen();
+        final String id = IndexFileNames.stripSegmentName(name);
+        final FileEntry entry = entries.get(id);
+        if (entry == null) {
+            String datFileName = IndexFileNames.segmentFileName(segmentName, "", LuceneOptimizedCompoundFormat.DATA_EXTENSION);
+            throw new FileNotFoundException("No sub-file with id " + id + " found in compound file \"" + datFileName + "\" (fileName=" + name + " files: " + entries.keySet() + ")");
+        }
+        return handle.slice(name, entry.offset, entry.length);
+    }
+
+    /** Returns an array of strings, one for each file in the directory. */
+    @Override
+    public String[] listAll() {
+        ensureOpen();
+        String[] res = entries.keySet().toArray(new String[entries.size()]);
+
+        // Add the segment name
+        for (int i = 0; i < res.length; i++) {
+            res[i] = segmentName + res[i];
+        }
+        return res;
+    }
+
+    /** Not implemented.
+     * @throws UnsupportedOperationException always: not supported by CFS */
+    @Override
+    public void deleteFile(String name) {
+        throw new UnsupportedOperationException();
+    }
+
+    /** Not implemented.
+     * @throws UnsupportedOperationException always: not supported by CFS */
+    @Override
+    public void rename(String from, String to) {
+        throw new UnsupportedOperationException();
+    }
+
+    /** Not implemented. */
+    @Override
+    public void syncMetaData() {
```

Comment:
Sonarqube warned ```Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation.``` here. This class is an optimized one as a replacement of ```Lucene50CompoundReader```, which also has an empty ```syncMetaData()``` implementation here. I don't think we want to fix this as suggested by Sonarqube.

Reply:
Just add a nested comment then?